src/provider/model/AppDirectory.ts

```diff
@@ -35,6 +37,12 @@ export class AppDirectory extends AsyncInit {
         }) || null;
     }
 
+    public async getAppByUuid(uuid: string): Promise<Application | null> {
+        return this._directory.find((app: Application) => {
+            return checkCustomConfigField(app, CustomConfigFields.OPENFIN_APP_UUID) === uuid || app.appId === uuid;
```

Comment:
It's still the case that there's potentially two input `uuids` that'll give the same app.

My claim is that if an appId is specified in the custom data, the other appId should be ignored entirely here. So something like

    const customValue = checkCustomConfigField(app, CustomConfigFields.OPENFIN_APP_UUID);
    return (customValue !== undefined)  ? (customValue === uuid) : (app.appId === uuid)

would work. And I think it'd be worth explicitly testing that if a directory entry specifies a UUID as a custom data field, appId gets igored by `getAppByUuid`